src/webgpu/api/operation/render_pass/clear_value.spec.ts

```diff
@@ -32,3 +38,121 @@ shaders and then checking their computed bind group layouts are compatible with
+    const { stencilFormat, stencilClearValue } = t.params;
+
+    const kSize = [1, 1, 1] as const;
+    const colorFormat = 'rgba8unorm';
+    const stencilTexture = t.device.createTexture({
+      format: stencilFormat,
+      size: kSize,
+      usage: GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    const colorTexture = t.device.createTexture({
+      format: colorFormat,
+      size: kSize,
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+    });
+    const renderPipeline = t.device.createRenderPipeline({
+      layout: 'auto',
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+            @vertex
+            fn main(@builtin(vertex_index) VertexIndex : u32)-> @builtin(position) vec4<f32> {
+              var pos : array<vec2<f32>, 6> = array<vec2<f32>, 6>(
+                  vec2<f32>(-1.0,  1.0),
+                  vec2<f32>(-1.0, -1.0),
+                  vec2<f32>( 1.0,  1.0),
+                  vec2<f32>(-1.0, -1.0),
+                  vec2<f32>( 1.0,  1.0),
+                  vec2<f32>( 1.0, -1.0));
+              return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: t.device.createShaderModule({
+          code: `
+            @fragment
+            fn main() -> @location(0) vec4<f32> {
+              return vec4<f32>(0.0, 1.0, 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [{ format: colorFormat }],
+      },
+      depthStencil: {
+        format: stencilFormat,
+        depthCompare: 'always',
+        stencilFront: {
+          compare: 'equal',
+        },
+        stencilBack: {
+          compare: 'equal',
+        },
+      },
+      primitive: {
+        topology: 'triangle-list',
+      },
+    });
+
+    const stencilAspectSizeInBytes = depthStencilFormatAspectSize(stencilFormat, 'stencil-only');
+    assert(stencilAspectSizeInBytes > 0);
+    const stencilReference = stencilClearValue & ((stencilAspectSizeInBytes << 8) - 1);
+
+    const encoder = t.device.createCommandEncoder();
+
+    const depthStencilAttachment: GPURenderPassDepthStencilAttachment = {
+      view: stencilTexture.createView(),
+      stencilLoadOp: 'clear',
+      stencilStoreOp: 'store',
+      stencilClearValue,
+    };
+    if (kTextureFormatInfo[stencilFormat].depth) {
+      depthStencilAttachment.depthClearValue = 0;
+      depthStencilAttachment.depthLoadOp = 'clear';
+      depthStencilAttachment.depthStoreOp = 'store';
+    }
+    const renderPassEncoder = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: colorTexture.createView(),
+          loadOp: 'clear',
+          storeOp: 'store',
+          clearValue: [1, 0, 0, 1] as const,
+        },
+      ],
+      depthStencilAttachment,
+    });
+    renderPassEncoder.setPipeline(renderPipeline);
+    renderPassEncoder.setStencilReference(stencilReference);
```

Comment:
I would like to have a case of setting stencil reference to  `stencilClearValue` rather than `stencilReference`, and expect failure if `stencilClearValue !== stencilReference`.

Reply:
Done.

I've also verified the content in the stencil aspect by copying the stencil data into a buffer. PTAL, thanks!